corehq/apps/reports/views.py

```diff
@@ -2032,27 +2032,38 @@ def _is_location_safe_report_class(view_fn, request, domain, export_hash, format
 @login_and_domain_required
 @require_GET
 def export_report(request, domain, export_hash, format):
-    cache = get_redis_client()
-    report_class = cache.get(export_hash)
     db = get_blob_db()
-    report_file = db.get(export_hash)
-    if report_file is not None:
-        if not request.couch_user.has_permission(domain, 'view_report', data=report_class):
-            raise PermissionDenied()
-        if format in Format.VALID_FORMATS:
-            file = ContentFile(report_file.read())
-            response = HttpResponse(file, Format.FORMAT_DICT[format])
-            response['Content-Length'] = file.size
-            response['Content-Disposition'] = 'attachment; filename="{filename}.{extension}"'.format(
-                filename=export_hash,
-                extension=Format.FORMAT_DICT[format]['extension']
-            )
-            return response
-        else:
-            return HttpResponseNotFound(_("We don't support this format"))
+
+    from corehq.blobs.models import BlobMeta
+    from corehq.blobs import NotFound
```

Comment:
Nit: is there a reason these were not put at the top of the module? Also, standard practice is to put imports first before any other code in the function (or module) unless circular dependencies prevent it.

Reply:
No reason, I had put it here while debugging and forgot to move it: https://github.com/dimagi/commcare-hq/pull/23927/commits/c04dca76224e94d3a288b233070755aaf7547101